cp: modify archive flag to copy dir contents rather than dir

dmatos2012 opened the pull request:
Hello,
This PR fixes #3886 to handle copying source's directory contents rather than the directory itself. 

Its my first ever Rust PR. Thanks for the helpful responses, and in case something else needs to be added, i can do so np.

Thanks:)

sylvestre commented:
I guess you saw that it failed on some other archs:

```

---- test_cp::test_cp_archive_on_directory_ending_dot stdout ----
current_directory_resolved: 
mkdir: C:\Users\RUNNER~1\AppData\Local\Temp\.tmphZwpwf\dir1
mkdir: C:\Users\RUNNER~1\AppData\Local\Temp\.tmphZwpwf\dir2
touch: C:\Users\RUNNER~1\AppData\Local\Temp\.tmphZwpwf\dir1/file
run: D:\a\coreutils\coreutils\target\debug\coreutils.exe cp -a dir1/. dir2
thread 'test_cp::test_cp_archive_on_directory_ending_dot' panicked at 'Command was expected to succeed.
stdout = 
 stderr = cp: XAttrs are only supported on unix.
', tests\common\util.rs:176:9


failures:
    test_cp::test_cp_archive_on_directory_ending_dot
```

dmatos2012 commented:
Thanks for the quick feedback. I just noticed, however I also ran the same command `cp -a dir1/. dir2`(on windows) on the main branch and I get the same failure. Thus, I think this behavior was present from before, but it was exposed via the test.

That being said, I dont have the knowledge so far to know how to make it work for windows(given the lack of `Xattrs`). So if maybe you can give me some pointers, I can perhaps read up on my own, and propose a fix on different PR(or this one as well)

Thanks again

sylvestre commented:
maybe it could be a unix test only ?

dmatos2012 commented:
Yeah, nice solution. Done :)

dmatos2012 commented:
Is there anything I shall do to get that final test passing? It looks like its unrelated to this PR. I am just making sure I have addressed the issues. Let me know :)

tertsdiepraam commented:
No need to worry about that test. The android build is failing on all PRs at the moment.

jtracey commented:
That failed test actually not because of the spurious failures from before, it's because we still need to decide how to fix #3477.

sylvestre commented:
@dmatos2012 sorry but it needs to be rebased

dmatos2012 commented:
uff I had completely forgotten about this PR :P @sylvestre , but managed to rebase :)

tertsdiepraam commented on src/uu/cp/src/copydir.rs:
```diff
@@ -93,7 +93,7 @@ impl<'a> Context<'a> {
         let current_dir = env::current_dir()?;
         let root_path = current_dir.join(root);
-        let root_parent = if target.exists() {
+        let root_parent = if target.exists() && !root.to_str().unwrap().ends_with("/.") {
```
Is this the only edge case in which the files are copied?

dmatos2012 replied:
I thought so from the original issue, but was so long ago I dont remember if there was something else present at the time :stuck_out_tongue:

tertsdiepraam replied:
Oh yeah it is super long ago 😄

tertsdiepraam replied:
Well, looks good. Sorry it took so long!
